apps/desktop/src/lib/commits/service.ts

```diff
@@ -0,0 +1,15 @@
+import { invoke } from '$lib/backend/ipc';
+import { Commit } from '$lib/vbranches/types';
+import { plainToInstance } from 'class-transformer';
+
+export class RemoteCommitService {
+	async find(projectId: string, commitOid: string): Promise<Commit | undefined> {
+		try {
+			const commit = await invoke<unknown>('find_commit', { projectId, commitOid });
+			return plainToInstance(Commit, commit);
+		} catch (error) {
+			console.error(error);
+			return undefined;
+		}
+	}
+}
```

Comment:
Something in my mind here is that we want to move away from using `console.error` and silently consuming errors. I figure here your intention was to represent a commit not being found, but I think with a little bit of jiggery pokery we can make it a little nicer.

```suggestion
import { invoke } from '$lib/backend/ipc';
import { Commit } from '$lib/vbranches/types';
import { plainToInstance } from 'class-transformer';
import type { Project } from '$lib/backend/projects';

export class CommitService {
	constructor(private project: Project) {}

	async find(commitOid: string): Promise<Commit> {
		const commit = await this.findMaybe(commitOid);

		if (!commit) {
			throw new Error(`Commit ${commitOid} not found`);
		}

		return commit;
	}

	async findMaybe(commitOid: string): Promise<Commit | undefined> {
		const commit = await invoke<unknown | undefined>('find_commit', { projectId: this.project.id, commitOid });

		if (!commit) return;

		return plainToInstance(Commit, commit);
	}
}
```

Reply:
We also know that the name "RemoteCommitSerice" is inaccurate, could we please call it "CommitService"

Reply:
I'd be keen to hear your thoughts on this, but in my mind it makes sense to try and pass as many dependencies in the constructor as possible. I would be keen to pass the project through the constructor until we had a good reason as to why that was a bad pattern, as it just adds more boilerplate to the consumers.

Reply:
Will do the changes 👍 

> in my mind it makes sense to try and pass as many dependencies in the constructor as possible
>

This goes into a nice conversation I'd like to have about at some point going in the other direction :).

Short answer: Yes, I'll change it to make the CommitService project-scoped. I think that will follow the pattern that we have in other services more closely.

Long-and-to-be-had-in-person-over-a-coffee answer: I think as we keep adding functionalities we might want to walk away from OOP patterns (where we can). 
In this case, for example, this could have been a simple function, that would be easily testable and modular.
I think in the long run we'll benefit from making the FE simpler where we can.
But would be interested to hear what you think :)

Reply:
> Long-and-to-be-had-in-person-over-a-coffee answer: I think as we keep adding functionalities we might want to walk away from OOP patterns (where we can).
In this case, for example, this could have been a simple function, that would be easily testable and modular.
I think in the long run we'll benefit from making the FE simpler where we can.
But would be interested to hear what you think :)

I'm rather keen to have this conversation. As someone who spent three months of his life learning Haskell, I'm partial to a bit of functional programming.